src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -2311,6 +2311,73 @@ public async Task responses_to_named_requests_can_be_accessed_as_symbols_in_late
         secondResult.Events.Should().NotContainErrors();
     }
 
+    [Fact]
+    public async Task Response_headers_for_named_requests_can_be_accessed_correctly()
+    {
+
+        var responseHandler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            response.RequestMessage = message;
+            var contentString = @"
+             {
+    ""headers"": {
+        ""Accept"": ""text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"",
+        ""Accept-Encoding"": ""gzip, deflate, br, zstd"",
+        ""Accept-Language"": ""en-US,en;q=0.9"",
+        ""Host"": ""httpbin.org"",
+        ""Priority"": ""u=0, i"",
+        ""Sec-Ch-Ua"": ""\""Chromium\"";v=\""128\"", \""Not;A=Brand\"";v=\""24\"", \""Microsoft Edge\"";v=\""128\"""",
+        ""Sec-Ch-Ua-Mobile"": ""?0"",
+        ""Sec-Ch-Ua-Platform"": ""\""Windows\"""",
+        ""Sec-Fetch-Dest"": ""document"",
+        ""Sec-Fetch-Mode"": ""navigate"",
+        ""Sec-Fetch-Site"": ""none"",
+        ""Sec-Fetch-User"": ""?1"",
+        ""Upgrade-Insecure-Requests"": ""1"",
+        ""User-Agent"": ""Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0"",
+        ""X-Amzn-Trace-Id"": ""Root=1-66e9f16e-7afea6f643e754f854c57d85""
+    }
+}
+            ";
+            response.Content = new StringContent(contentString, Encoding.UTF8, "application/json");
+            response.Headers.Add("server", "gunicorn/19.9.0");
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(responseHandler);
+        // var client = new HttpClient();
+        using var kernel = new HttpKernel("http", client);
+
+        var code = """
+            @baseUrl = https://httpbin.org/headers
+
+            # @name binHeader
+            GET {{baseUrl}}
+
+            ###
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+
+        var secondCode = $$$"""
+            GET https://httpbin.org/headers
+            Detectedserver: {{binHeader.response.headers.Server}}
```

Comment:
As it stands the test does not validate the binding for this named request expression (since the message handler always returns the same value).

Should this test somehow validate that the value for the `Detectedserver` header matches the expected value `gunicorn/19.9.0` here? One way to do that may be to update the message handler such that it puts the bound value of any header with name `Detectedserver` in the body of the returned response. The test could then assert that the response body has the correct value...

Reply:
I often take that approach. It can be done as well by just capturing the bound header within the message handler and setting a variable to assert against in the tests's outer scope. I think this is a little more direct.